src/app/hooks/useCarts.tsx

```diff
@@ -40,7 +40,7 @@ export const CartContextProvider = (props: Props)=> {
         localStorage.removeItem('cartProduct')
         success('product removed succesfully')
       }
-   }, [cartProducts, success])
+   }, [])
```

Comment:
Same as this https://github.com/Kasia-Africa/kasia-frontend/pull/1/files#r1485210631, in your useCallback, you are using product and you also have a success function but you failed to declare those dependencies here as well. Replace `[]` with `[cartProducts, success]`